pkg/types/clusterconfig/clusterconfig.go

```diff
@@ -1078,6 +1079,58 @@ func DefaultAccessConfig() (*AccessConfig, error) {
 	return accessConfig, nil
 }
 
+func (cc *Config) MaxPossibleOnDemandInstances() int64 {
+	if cc.MaxInstances == nil {
+		return 0 // unexpected
+	}
+
+	if cc.Spot == nil || *cc.Spot == false || cc.SpotConfig == nil || cc.SpotConfig.OnDemandBackup == nil || *cc.SpotConfig.OnDemandBackup == true {
+		return *cc.MaxInstances
+	}
+
+	var count int64
+
+	// default OnDemandBaseCapacity is 0
+	if cc.SpotConfig.OnDemandBaseCapacity != nil {
+		count += *cc.SpotConfig.OnDemandBaseCapacity
+	}
+
+	// default OnDemandPercentageAboveBaseCapacity is 0
+	if cc.SpotConfig.OnDemandPercentageAboveBaseCapacity != nil {
+		count += int64(math.Ceil(float64(*cc.SpotConfig.OnDemandPercentageAboveBaseCapacity) / 100 * float64(*cc.MaxInstances-count)))
```

Comment:
Just to prevent bugs, I would use `cc.SpotConfig.OnDemandBaseCapacity` directly instead of using count as a variable that temporarily stores `cc.SpotConfig.OnDemandBaseCapacity` `float64(*cc.MaxInstances-count)`. Maybe in L1095 define a separate variable specifically for `cc.SpotConfig.OnDemandBaseCapacity` and use that variable here.